utils/db.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Contain methods for persistence functonality
+module Database
+  private
+
+  # All the data of the library can be to stored/loaded into some storage
+  # (YML,JSON etc)
+  def save_data
+    data = {
+      authors: @authors,
+      books: @books,
+      orders: @orders,
+      readers: @readers
+    }
+
+    File.write(Config::DB_PATH + Config::DB_FILE, data.to_yaml)
+  end
+
+  # All the data of the library can be to stored/loaded into some storage
+  # (YML,JSON etc)
+  def load_data
+    yaml = File.read(Config::DB_PATH + Config::DB_FILE)
+    data = Psych.safe_load(
+      yaml, [Symbol, Date, Author, Book, Reader, Order], aliases: true
+    )
+
+    @authors  = data[:authors]
+    @books    = data[:books]
+    @orders   = data[:orders]
+    @readers  = data[:readers]
+  end
+
+  # Generating random data using Faker gem
+  def generate_data
+    Config::ENTITIES.each do |entity|
```

Comment:
Code with metaprogramming looks shorter, but it's less readable. I would better do smth like this:
```
data = FakeDataGenerator.call # returns a hash {authors: [], books: [], orders: [], readers: []}
authors  = data[:authors]
books    = data[:books]
orders   = data[:orders]
readers  = data[:readers]
```
Also you could extract common code (`authors  = data[:authors]`...) from `load_data` and `generate data` to a separate method.
You don't have to change this, just my thoughts